stdlib/messaging/kafka/src/main/ballerina/src/kafka/producer.bal

```diff
@@ -174,13 +199,174 @@ public type Producer client object {
     # + partition - Partition to which the record should be sent.
     # + timestamp - Timestamp of the record, in milliseconds since epoch.
     # + return - Returns `kafka:ProducerError` if send action fails to send data, nil otherwise.
-    public remote function send(byte[] value, string topic, public byte[]? key = (), public int? partition = (),
-                                public int? timestamp = ()) returns ProducerError? {
-        return producerSend(self, value, java:fromString(topic), key, partition, timestamp);
+    public remote function send(Data value, string topic, public Data? key = (), public int? partition = (),
+                                                            public int? timestamp = ()) returns ProducerError? {
```

Comment:
Shall we change indentations as previous?

Reply:
There are a few styling issues and also the java methods needed to re-arrange after the inter-op migration. This will be addressed in that PR.